### DISCUSSION TITLE
Two local searx instances: settings are messed up

### DISCUSSION THREAD

**Discussion by Twix53791** [searxng/searxng#3339] [Q&A]
**OS: arch linux**
**searx: ** just installed from github with the install script searxng.sh

Hi, I am tring to run on my local machine two searx instances with two different languages : one in english, one in french. I would like to set two differents settings.yml.

I run two services : `uwsgi@searxng.service` and `uwsgi@searxfr.service`.

journalctl -xeu uwsgi@searxng.service gives:
```
A start job for unit uwsgi@searxng.service has begun execution.
...
uwsgi[19893]: [uWSGI] getting INI configuration from /etc/uwsgi/searxng.ini
```

journalctl -xeu uwsgi@searxfr.service gives:
```
A start job for unit uwsgi@searxfr.service has begun execution.
...
uwsgi[20272]: [uWSGI] getting INI configuration from /etc/uwsgi/searxfr.ini
```

searxng.ini:
```
env = SEARXNG_SETTINGS_PATH=/etc/searxng/settingsen.yml
...
http = 127.0.0.1:8888
```

searxfr.ini:
```
env = SEARXNG_SETTINGS_PATH=/etc/searxng/settingsfr.yml
...
http = 127.0.0.2:8888
```

settingsen.yml
```
use_default_settings: true

general:
  debug: false
  instance_name: "SearXNG"

search:
  safe_search: 0
  autocomplete: 'duckduckgo'
  default_lang: 'en-US'
  languages:
    - all
    - en
    - en-US
    - en-GB
    - it-IT
    - fr-FR
    - de
    - es
    - ru-RU
 ```

settingsfr.yml:
```
use_default_settings: true

general:
  debug: false
  instance_name: "SearXNG"

search:
  safe_search: 0
  autocomplete: 'duckduckgo'
  default_lang: 'fr-FR'
  languages:
    - en
    - en-US
    - fr
    - fr-FR
```

I use nginx to run the two servers. It works, I ve got 127.0.0.1 and 127.0.0.2, but the settings are messed up. Actually, if I run the searxfr.service last, both instances have the settingsfr.yml settings, and if I run the searxng.service last, the contrary...

Thanks for any clue!

**Comment by unixfox** [comment]
you are probably going to have difficulties as the install script was not meant to handle such cases.

maybe if you use docker you could do it more easily.

**Reply by Twix53791** [reply]
Hi, well, until now I avoided to use docker in all my linux set up, I would like to find a way. Actually, on firefox, it is possible to handle two different preferences set in the browser for the server 127.0.0.1 and 127.0.0.2, and I can set different language in the preferences. For now, in qutebrowser, my main browser, this doesn't work, I am gonna try to find why...

**Reply by Twix53791** [reply]
maybe there is a way to define different cookies for two different uwsgi instances? 
or maybe there is another way to use an alternate language? In fact, my goal is really simple : be able to switch very easilly from one language to another, using currently the qutebrowser bang system: ! search run the searxfr when the normal search run the searxng, default one in english.

**Reply by Twix53791** [reply]
UPDATE : 

I found a way... I fix my qutebrowser problem (I had to accept all cookies without blocking third party cookies). 
I abadonned the idea to have different settings.yml files : now all the uwsgi services use the same setttings.yml file.
The languages are set with the cookies of the two instances, wich are not the same has the ip adress (and not only the port) is different. It works fine.
